qcodes/dataset/data_set.py

```diff
@@ -60,66 +60,79 @@ class CompletedError(RuntimeError):
     pass
 
 
-class Subscriber(Thread):
+class _Subscriber(Thread):
     """
     Class to add a subscriber to a DataSet. The subscriber gets called
     every time an insert is made to the results_table.
 
-    The Subscriber is not meant to be instantiated directly, but rather used
+    The _Subscriber is not meant to be instantiated directly, but rather used
     via the 'subscribe' method of the DataSet.
 
-    NOTE: A subscriber should be added *after* all parameters have added.
+    NOTE: A subscriber should be added *after* all parameters have been added.
+
+    NOTE: Special care shall be taken when using the *state* object: it is the user's
+    responsibility to operate with it in a thread-safe way.
     """
-    def __init__(self, dataSet, sub_id: str,
+    def __init__(self,
+                 dataSet,
+                 id_: str,
                  callback: Callable[..., None],
-                 state: Optional[Any] = None, min_wait: int = 100,
-                 min_count: int = 1,
-                 callback_kwargs: Optional[Dict[str, Any]]=None) -> None:
-        self.sub_id = sub_id
-        # whether or not this is actually thread safe I am not sure :P
+                 state: Optional[Any] = None,
+                 loop_sleep_time: int = 0,  # in seconds
+                 min_queue_length: int = 1,
+                 callback_kwargs: Optional[Dict[str, Any]]=None
+                 ) -> None:
+        super().__init__()
+
+        self._id = id_
+
         self.dataSet = dataSet
         self.table_name = dataSet.table_name
-        conn = dataSet.conn
-        self.log = logging.getLogger(f"Subscriber {self.sub_id}")
+        self._data_set_len = len(dataSet)
 
         self.state = state
-        self.min_wait = min_wait
-        self.min_count = min_count
-        self._send_queue: int = 0
+
+        self.data_queue: Queue = Queue()
+        self._queue_length: int = 0
+        self._stop_signal: bool = False
+        self._loop_sleep_time = loop_sleep_time / 1000  # convert seconds to minutes
```

Comment:
I think you mean convert milli-seconds to seconds

Reply:
absolutely correct, will fix it

Reply:
done